src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -124,6 +124,19 @@ public async Task<ImmutableArray<ChangeSignatureCodeAction>> GetChangeSignatureC
                 {
                     symbol = typeSymbol.DelegateInvokeMethod;
                 }
+                else if (typeSymbol.IsRecord)
+                {
+                    // A bit hacky to determine the parameters of primary constructor associated with a given record.
+                    // TODO: record structs.
+                    const int RecordDeclarationRawKind = 9063;
+                    var potentialPrimaryCtor = typeSymbol.InstanceConstructors.FirstOrDefault(
```

Comment:
Hmm, I'm a little worried about hardcoding the raw kind here. I almost think we should wait on merging until we have a proper workaround that doesn't involve hardcoding. @CyrusNajmabadi do you have any thoughts?

Reply:
I think https://github.com/dotnet/roslyn/issues/53092 may not be very soon 😕

Reply:
we're in an abstract type, just provide a helper that C# implements, going back to source to answer this.